views/explore/Random event - meet girl #3.tw

```diff
@@ -1,28 +1,156 @@
 :: Random event - meet girl #3 {"position":"5742,16","size":"100,100"}
 <h1 class="ptitle">FOREST</h1>
 <br /><br />
-<center>
- [img[setup.ImagePath+'places/explore/found_girl.jpg']]
- </center>
- <br /><br />
-You notice a girl walking through the forest and decide to check out what she's doing.
- <br /><br />
+<<set $tmpGirl = null, $slaveId = null, $tempGuy = null>>
 
-<<link 'Go closer'>>
+<<if (settings.orientationMC is 'hetero')>>
+			<<set _probGirl to 100>>
+			<<if (settings.gaysEnabled)>>	
+					<<set _probGirl to 90>>
+			<</if>>
+			<<if (settings.transEnabled)>>
+					<<set _probGirl = 94>><<set _probTGirl = 97>>
+			<</if>>
+			<<if (settings.transEnabled &amp;&amp; settings.gaysEnabled)>>
+					<<set _probGirl = 84>><<set _probBoy = 94>><<set _probTGirl = 97>>
+			<</if>>
+<</if>>
+<<if (settings.orientationMC is 'bi')>>
+			<<set _probGirl = 90>>
+			<<if (settings.gaysEnabled)>>	
+					<<set _probGirl = 50>>
+			<</if>>
+			<<if (settings.transEnabled)>>
+					<<set _probGirl = 94>><<set _probTGirl = 97>>
```

Comment:
And there is a lot of copy/paste code.
First part looks very similar, so probably could just create an object where would be defined per orientation.
For example
__mapping = {
bi: {
girl: 90,
probTGirl:97,
probBoy:94
}
};

etc